src/main/kotlin/blackjack/controller/Controller.kt

```diff
@@ -0,0 +1,123 @@
+package blackjack.controller
+
+import blackjack.model.*
+import blackjack.view.InputView
+import blackjack.view.OutputView
+
+class Controller() {
+    private var players: List<Gambler> = mutableListOf()
+    private val dealer: Dealer = Dealer(GamblerInfo("dealer"))
+    private val deck = Deck()
+
+    fun run() {
+        try {
+            createPlayers()
+            firstDraw()
+            finalDraw()
+            showResults(FinalResult(dealer, players))
+        } catch (err: IllegalArgumentException) {
+            OutputView.displayErrorMessages(err.message)
+        }
+    }
+
+    private fun createPlayers() {
+        players =
+            processPlayerNames().map { Gambler(it) }
+        OutputView.displayNamesOfPlayers(players)
+    }
+
+    private fun firstDraw() {
+        dealer.addCard(getInitialCards())
+        players.forEach { it.addCard(getInitialCards()) }
+        OutputView.displayCardsOfDealer(dealer)
+        OutputView.displayCardsOfPlayers(players)
+    }
+
+    private fun finalDraw() {
+        players.forEach { playerTakesTurn(it) }
+        dealerTakesTurn()
+        OutputView.displayCardsOfPlayersWithScore(listOf(dealer) + players)
+        OutputView.displayFinalResultsHeading()
+    }
+
+    private fun getInitialCards(): List<Card> = deck.drawCards(INITIAL_CARD_COUNT)
+
+    private fun dealerTakesTurn() {
+        while (dealer.isDealerBelowMinScore()) {
+            dealer.addCard(deck.drawCards())
+            OutputView.displayDealersTurn()
+        }
+    }
+
+    private fun playerTakesTurn(player: Gambler) {
+        var answer = false
+        while (player.isPlayerBelowBlackJack()) {
+            answer = processHitOrStay(player)
+            if (!answer) {
+                break
+            }
+            player.addCard(deck.drawCards())
+            OutputView.displayCardsOfPlayers(player)
+        }
+        if (!answer && player.cards.size == INITIAL_CARD_COUNT) {
+            OutputView.displayCardsOfPlayers(player)
+        }
+    }
+
+    private fun showResults(finalResult: FinalResult) {
+        OutputView.displayPlayerResult(
+            finalResult.lose.size,
+            finalResult.win.size,
+            finalResult.draw.size,
+        )
+        finalResult.win.forEach {
+            OutputView.displayPlayerResult(it.name, true)
+        }
+        finalResult.lose.forEach {
+            OutputView.displayPlayerResult(it.name, false)
+        }
+        finalResult.draw.forEach {
+            OutputView.displayPlayerResult(it.name, false)
+        }
+    }
+
+    private fun processPlayerNames(): List<GamblerInfo> {
+        repeat(MAX_ATTEMPTS) {
+            try {
+                val names = InputView.getNamesOfPlayers()
+                return names
+                    .parseCommaString()
+                    .map(::GamblerInfo)
```

Comment:
Nice job refactoring the logic.
Excellent use of an extension function as well. 👍